### PR TITLE
Add meta field to predicate signature to avoid computing the same things multiple times

### PR DESCRIPTION
**Pull request by wojtek-t**
This PR only uses it to avoid computing QOS of a pod for every node from scratch.

Ref #28590

### CODE REVIEW

**Comment by wojtek-t** [comment]
@davidopp - PTAL
[I'm going to take even more advantage of it in the following PRs.]

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -65,6 +65,22 @@ func (c *CachedNodeInfo) GetNodeInfo(id string) (*api.Node, error) {
 	return node.(*api.Node), nil
 }
 
+// podMetadata defines a type, that is an expected type that is passed
```
Comment (davidopp) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: you could just say "podMetadata is a type that is passed as metadata for predicate functions"

**Comment by davidopp** [comment]
LGTM

Feel free to fix the comment thing in a follow-up PR, or fix it in this one and re-add LGTM label yourself.

**Comment by davidopp** [comment]
@k8s-bot test this issue: #28683

**Comment by davidopp** [comment]
@k8s-bot test this issue: #28275

**Comment by davidopp** [comment]
"reference is not a tree"

**Comment by davidopp** [comment]
@k8s-bot test this issue: #IGNORE

**Pull request merged**